stacker/status.py

```diff
@@ -4,10 +4,17 @@ def __init__(self, name, code, reason=None):
         self.code = code
         self.reason = reason or getattr(self, "reason", None)
 
+    def cmp(self, a, b):
+        try:
+            return cmp(a, b)
+        except NameError:
+            # Python3 doesn't have cmp function.
+            return ((a > b) - (a < b))
+
     def __cmp__(self, other):
         if hasattr(other, "code"):
-            return cmp(self.code, other.code)
-        return False
+            return self.cmp(self.code, other.code)
+        raise Exception("Both Status objects must have a `code` attribute.")
```

Comment:
The correct thing to do here is to ```return NotImplemented``` (and usually if the types are different too). In python2 this will propagate to the default logic of an identity comparison, which is at least internally consistent, and is definitely better than accidentally returning unconditional equality via ```False``` meaning "no difference". In python3 however this code won't even get called as ```__cmp__``` is no longer a magic method but that can be left for the py3 branch.

As before my reason for raising the issue and not just steamrolling over it was concern over the possibility of the ordering of these things mattering somewhere and this bug's behavior subtlely being actually relied upon, but as that isn't the case any correction is an improvement.